_aggregations/pipeline/bucket-selector.md

```diff
@@ -0,0 +1,123 @@
+| `gap_policy`          | Optional          | String          | The policy to apply to missing data. Valid values are `skip`, `insert_zeros`, and `keep_values`. Default is `skip`. |
+
+
+## Example
+
+The following example creates a date histogram with a one-week interval from the OpenSearch Dashboards e-commerce sample data. The `sum` subaggregation calculates the sum of all sales for each week. Finally, the `bucket_selector` aggregation filters the resulting weekly buckets, removing all the buckets that do not total more than $75,000:
+
+```json
+GET opensearch_dashboards_sample_data_ecommerce/_search
+{
+  "size": 0,
+  "aggs": {
+    "sales_per_week": {
+      "date_histogram": {
+        "field": "order_date",
+        "calendar_interval": "week"
+      },
+      "aggs": {
+        "weekly_sales": {
+          "sum": {
+            "field": "taxful_total_price",
+            "format": "$#,###.00"
+          }
+        },
+        "avg_vendor_spend": {
+          "bucket_selector": {
+            "buckets_path": {
+              "weekly_sales": "weekly_sales"
+            },
+            "script": "params.weekly_sales > 75000"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+The aggregation returns the `sales_per_week` buckets that meet the scripted criterion:
+
+```json
+{
+  "took": 3,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 4675,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "aggregations": {
+    "sales_per_week": {
+      "buckets": [
+        {
+          "key_as_string": "2025-03-31T00:00:00.000Z",
+          "key": 1743379200000,
+          "doc_count": 1048,
+          "weekly_sales": {
+            "value": 79448.60546875,
+            "value_as_string": "$79,448.61"
+          }
+        },
+        {
+          "key_as_string": "2025-04-07T00:00:00.000Z",
+          "key": 1743984000000,
+          "doc_count": 1048,
+          "weekly_sales": {
+            "value": 78208.4296875,
+            "value_as_string": "$78,208.43"
+          }
+        },
+        {
+          "key_as_string": "2025-04-14T00:00:00.000Z",
+          "key": 1744588800000,
+          "doc_count": 1073,
+          "weekly_sales": {
+            "value": 81277.296875,
+            "value_as_string": "$81,277.30"
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+Because it returns a Boolean rather than a numeric value, the `buckets_selector` aggregation does not take a `format` parameter. In this example, the formatted metrics are returned in the `value_as_string` result by the `sum` sub-aggregation. Contrast this with the example in the [`bucket_script` aggregation]({{site.url}}{{site.baseurl}}/aggregations/pipeline/bucket-script/).
```

Comment:
```suggestion
Because it returns a Boolean rather than a numeric value, the `buckets_selector` aggregation does not take a `format` parameter. In this example, the formatted metrics are returned in the `value_as_string` result by the `sum` subaggregation. Contrast this with the example in the [`bucket_script` aggregation]({{site.url}}{{site.baseurl}}/aggregations/pipeline/bucket-script/).
```